pandas/core/arrays/categorical.py

```diff
@@ -2284,7 +2284,16 @@ def _replace(self, *, to_replace, value, inplace: bool = False):
         ser = ser.replace(to_replace=to_replace, value=value)
 
         all_values = Index(ser)
-        new_categories = Index(ser.drop_duplicates(keep="first"))
+
+        # GH51016: maintain order of existing categories
+        idxr = cat.categories.get_indexer_for(all_values)
```

Comment:
Do we need to do all this if `ordered=False`?

Reply:
1.5x maintains the existing order of categories regardless of `ordered` so I'd lean towards keeping that behavior. The new ordering logic is barely measurable perf-wise relative to the whole replace operation.